packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -2,61 +2,128 @@ import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
 import BigNumber from 'bignumber.js'
 import * as React from 'react'
-import { StyleSheet, Text, View } from 'react-native'
+import { StyleProp, StyleSheet, Text, TextStyle, View } from 'react-native'
+import { MoneyAmount } from 'src/apollo/types'
 import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
-import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { LocalCurrencyCode, LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { useDollarsToLocalAmount, useLocalCurrencyCode } from 'src/localCurrency/hooks'
 import { getMoneyDisplayValue } from 'src/utils/formatting'
 
+export enum DisplayType {
+  Default,
+  Big, // symbol displayed as superscript
+}
+
 interface Props {
-  amount: BigNumber
-  size: number
-  type: CURRENCY_ENUM
-  currencySymbol?: LocalCurrencySymbol | null
+  type: DisplayType
+  amount: MoneyAmount
+  size: number // only used for DisplayType.Big
+  useColors: boolean
+  hideSymbol: boolean
+  formatAmount: (amount: BigNumber.Value) => string
+  style?: StyleProp<TextStyle>
 }
 
-const symbolRatio = 0.6
+const SYMBOL_RATIO = 0.6
 
-// TODO(Rossy) This is mostly duped by MoneyAmount, converge the two
-export default class CurrencyDisplay extends React.PureComponent<Props> {
-  color() {
-    return this.props.type === CURRENCY_ENUM.DOLLAR ? colors.celoGreen : colors.celoGold
+function getBigSymbolStyle(fontSize: number, color: string) {
+  const size = Math.floor(fontSize * SYMBOL_RATIO)
+  return {
+    fontSize: size,
+    color,
+    lineHeight: Math.round(size * 1.4),
+    transform: [{ translateY: Math.round(size * 0.1) }],
   }
+}
 
-  symbolStyle(fontSize: number) {
-    const size = Math.floor(fontSize * symbolRatio)
-    return {
-      fontSize: size,
-      color: this.color(),
-      lineHeight: Math.round(size * 1.4),
-      transform: [{ translateY: Math.round(size * 0.1) }],
-    }
-  }
+// TODO(Rossy) This is mostly duped by MoneyAmount, converge the two
```

Comment:
Does this PR solve this TODO?

Reply:
Not entirely, I'll move existing `MoneyAmount` usage in a follow up PR related to a  more holistic solution for displaying currencies.